stdlib/number.gr

```diff
@@ -398,3 +398,77 @@ export let sin = (radians: Number) => {
 export let cos = (radians: Number) => {
   sin(pi / 2 + radians)
 }
+
+/**
+ * Computes the tan of a number (in radians) using Chebyshev polynomials.
+ *
+ * @param radians: The input in radians
+ * @returns The computed tangent
+ *
+ * @since v0.5.2
+ */
+export let tan = (radians: Number) => {
+  sin(radians) / cos(radians)
+}
+
+/**
+ * Computes the factorial of a number.
+ *
+ * @param input: The number to compute the factorial of
+ * @returns The computed factorial
+ *
+ * @since v0.5.2
+ */
+export let factorial = input => {
+  let s = sign(input)
+  let mut result = 1
+  for (let mut n = abs(input); n > 0; n -= 1) {
+    result *= n
+  }
+  if (s == 0) result else result * s
+}
+
+/**
+ * Clamps a number to a given range.
+ *
+ * @param min: The minimum value in the range
+ * @param max: The maximum value in the range
+ * @param input: The number to clamp
+ * @returns The clamped number
+ *
+ * @since v0.5.2
+ */
+export let clamp = (min: Number, max: Number, input: Number) => {
+  if (input < min) min else if (input > max) max else input
+}
+
+/**
+ * Linearly interpolates between the starting and ending value by the given amount.
+ *
+ * @param start: The starting value
+ * @param end: The ending value
+ * @param amount: an amount between 0 and 1
```

Comment:
This isn't enforced, so any amount outside of the range 0 to 1 produces an extrapolation. Would make sense to update this param description and the full description so that's clear.